<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Interactive Graph Cut Segmentation</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout" rowstretch="1,17,6">
      <item row="0" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Original Image</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Segmented Image</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QVTKWidget" name="qvtkWidgetLeft" native="true"/>
      </item>
      <item row="1" column="1">
       <widget class="QVTKWidget" name="qvtkWidgetRight" native="true"/>
      </item>
      <item row="2" column="0">
       <widget class="QGroupBox" name="grpSelection">
        <property name="styleSheet">
         <string notr="true">QGroupBox#grpSelection { 
     border: 2px solid gray; 
     border-radius: 3px; 
 } 

QGroupBox#grpSelection::title { 
    background-color: transparent;
 } </string>
        </property>
        <property name="title">
         <string>Selection properties</string>
        </property>
        <widget class="QRadioButton" name="radBackground">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>80</y>
           <width>106</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Background</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radForeground">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>50</y>
           <width>106</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Foreground</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnClearSelections">
         <property name="geometry">
          <rect>
           <x>220</x>
           <y>10</y>
           <width>211</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Clear All Selections</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnSaveSelections">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>140</y>
           <width>121</width>
           <height>41</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The foreground and background pixels will be saved as binary images in X_Foreground.png and X_Background.png where X is specified in the file save diaglog.</string>
         </property>
         <property name="text">
          <string>Save Selections</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>30</y>
           <width>171</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Current selection mode:</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnClearForeground">
         <property name="geometry">
          <rect>
           <x>220</x>
           <y>50</y>
           <width>211</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Clear Foreground Selections</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnClearBackground">
         <property name="geometry">
          <rect>
           <x>220</x>
           <y>90</y>
           <width>211</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Clear Background Selections</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnLoadSelections">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>140</y>
           <width>121</width>
           <height>41</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The foreground and background pixels will be saved as binary images in X_Foreground.png and X_Background.png where X is specified in the file save diaglog.</string>
         </property>
         <property name="text">
          <string>Load Selections</string>
         </property>
        </widget>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QGroupBox" name="groupBox">
        <property name="styleSheet">
         <string notr="true">QGroupBox#groupBox { 
     border: 2px solid gray; 
     border-radius: 3px; 
 } 

QGroupBox#groupBox::title { 
    background-color: transparent;
 } </string>
        </property>
        <property name="title">
         <string/>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>91</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>LambdaMax</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtLambdaMax">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>30</y>
           <width>41</width>
           <height>27</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The value of Lambda if the slider is set to 100%</string>
         </property>
         <property name="text">
          <string>.05</string>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>60</y>
           <width>131</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>% of LambdaMax</string>
         </property>
        </widget>
        <widget class="QSlider" name="sldLambda">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>80</y>
           <width>160</width>
           <height>19</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The percentage of LambdaMax to use as Lambda</string>
         </property>
         <property name="value">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="lblLambda">
         <property name="geometry">
          <rect>
           <x>220</x>
           <y>50</y>
           <width>51</width>
           <height>17</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The final value of Labmda to use. When Lambda is high, there is no regularization. That is, each pixel decides its label independently (does not care about the labeling of its neighbors).</string>
         </property>
         <property name="text">
          <string>.025</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>210</x>
           <y>30</y>
           <width>64</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Lambda</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>80</y>
           <width>41</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>100%</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>21</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>0%</string>
         </property>
        </widget>
        <widget class="QSlider" name="sldHistogramBins">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>130</y>
           <width>160</width>
           <height>19</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>The higher this value, the more discriminative the regional term will be.</string>
         </property>
         <property name="value">
          <number>40</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="lblHistogramBins">
         <property name="geometry">
          <rect>
           <x>240</x>
           <y>110</y>
           <width>41</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>20</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>110</y>
           <width>211</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Histogram Bins Per Dimension:</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnCut">
         <property name="geometry">
          <rect>
           <x>313</x>
           <y>10</y>
           <width>91</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string>Cut Graph</string>
         </property>
        </widget>
        <widget class="QProgressBar" name="progressBar">
         <property name="geometry">
          <rect>
           <x>303</x>
           <y>40</y>
           <width>118</width>
           <height>23</height>
          </rect>
         </property>
         <property name="value">
          <number>24</number>
         </property>
        </widget>
        <widget class="QLabel" name="label_7">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>130</y>
           <width>16</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>1</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>130</y>
           <width>21</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>99</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>150</y>
           <width>101</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>RGB Weight</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radDepthOnly">
         <property name="geometry">
          <rect>
           <x>300</x>
           <y>90</y>
           <width>116</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Depth Only</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QRadioButton" name="radColorOnly">
         <property name="geometry">
          <rect>
           <x>300</x>
           <y>120</y>
           <width>116</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Color Only</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radCombination">
         <property name="geometry">
          <rect>
           <x>300</x>
           <y>150</y>
           <width>116</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Combination</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="chkDebug">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>170</y>
           <width>131</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Debug Outputs</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="chkDepthHistogram">
         <property name="geometry">
          <rect>
           <x>246</x>
           <y>170</y>
           <width>181</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Use depth in histogram</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenImage"/>
    <addaction name="actionSaveSegmentation"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionFlipImage"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpenImage">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionOpen_Grayscale_Image">
   <property name="text">
    <string>Open As Grayscale Image</string>
   </property>
  </action>
  <action name="actionOpen_RGBDI_Image">
   <property name="text">
    <string>Open As RGBDI Image</string>
   </property>
  </action>
  <action name="actionSaveSegmentation">
   <property name="text">
    <string>Save Segmentation</string>
   </property>
  </action>
  <action name="actionFlipImage">
   <property name="text">
    <string>Flip Image</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>sldHistogramBins</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>lblHistogramBins</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>562</x>
     <y>805</y>
    </hint>
    <hint type="destinationlabel">
     <x>713</x>
     <y>783</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
